src/main/webapp/src/app/ranking/ranking-list/ranking-list.component.ts

```diff
@@ -15,20 +15,21 @@ export class RankingListComponent implements OnInit {
   hiddenGems! : HiddenGem[];
   location = {} as Location;
 
-  constructor(private hiddenGemService: HiddenGemService, private locationService: LocationService) { }
+  constructor(private locationService: LocationService, private appComponent: AppComponent) { }
 
   ngOnInit() {
-
     this.locationService.getLocation
       .subscribe(location => {
         this.location = location;
-    })
+      })
+    this.getRankedHiddenGems();
```

Comment:
Is this calling out to the server every time it's initialised or is there some caching I'm missing?

Reply:
I am not too sure I understand the question. What I thought when putting this function here is that it will be called once when the app is loading, and not be recalled unless the app is reloaded. Was there any other way I could it, that might be better than here?

Reply:
I thought the ngOnInit function ran every time the component was loaded, so when you switch back and forth between recommendations and the main page I would have thought this function would run again but I could be wrong.
You could test that by putting a log statement in there and try flipping back and forth.
I was imagining you'd bind the hidden gems array from the app.component to an input in this component, so the object is loaded once and passed around to all the sub components.
Either way this would be an optimisation so I'll let you submit for now, but maybe add that to the todo list in the project in case you are looking for more to do later.